### PR TITLE
Improvemen user model

### PR DESCRIPTION
**Pull request by qq1u**
## Description
The "_hash_digest " field is now deleted and the "password " field is used

### What issue is this solving?

Closes codeforpdx/dwellingly-app/issues/###

### CODE REVIEW

**Review by aedwardg — DISMISSED** [review]
This is nice, very pythonic!
Great job @13528080556!

File: tests/unit/test_user.py [hunk 1]
```diff
@@ -116,8 +116,7 @@ def test_update_class_method(self, create_join_staff, faker):
         UserModel.update(UserSchema, user.id, {"email": email})
         lookedup = UserModel.find(user.id)
         assert lookedup.email == email
-        assert lookedup.password is None
-        assert lookedup.hash_digest == user.hash_digest
+        assert lookedup.password == user.password
```
Comment (NickSchimek) [inline tests/unit/test_user.py]: I think we can delete this assertion. We're now no longer need to test behavior that we implemented. This is testing behavior provided by sqlalchemy.

File: models/user.py [hunk 2]
```diff
@@ -81,15 +86,8 @@ def validate_reset_password(token):
...
-    @staticmethod
-    def _hash_pw(plaintext_password):
-        return bcrypt.hashpw(
-            bytes(plaintext_password, "utf-8"),
-            bcrypt.gensalt(current_app.config["WORK_FACTOR"]),
-        )
-
     def check_pw(self, plaintext_password):
-        return bcrypt.checkpw(bytes(plaintext_password, "utf-8"), self.hash_digest)
+        return bcrypt.checkpw(plaintext_password.encode("utf-8"), self.password)
```
Comment (NickSchimek) [inline models/user.py]: I think I would rather see this: And then delete the password getter. As password should not be needed outside of this class.
```suggestion
        return bcrypt.checkpw(plaintext_password.encode("utf-8"), self._password)
```

File: models/user.py [hunk 3]
```diff
@@ -56,10 +55,16 @@ class UserModel(BaseModel):
         collection_class=NobiruList,
     )
 
-    def __init__(self, **kwargs):
-        super(UserModel, self).__init__(**kwargs)
-        self.hash_digest = UserModel._hash_pw(kwargs["password"])
-        self.archived = False
+    @property
+    def password(self):
+        return self._password
```
Comment (NickSchimek) [inline models/user.py]: I think this can be removed. It's not necessary to ever access the password outside of this object.

**Review by NickSchimek — CHANGES_REQUESTED** [review]
This looks great! Thanks for doing this.
I have a couple small suggestions. Let me know if they wont work for some reason.

File: models/user.py [hunk 4]
```diff
@@ -38,8 +38,7 @@ class UserModel(BaseModel):
     firstName = db.Column(db.String(100), nullable=False)
     lastName = db.Column(db.String(100), nullable=False)
     phone = db.Column(db.String(20), nullable=False)
-    hash_digest = db.Column(db.LargeBinary(60))
-    password = db.Column(db.String(), default=None, onupdate=None)
+    _password = db.Column("password", db.LargeBinary(60))
```
Comment (NickSchimek) [inline models/user.py:44]: What does this do `db.Column("password", db.LargeBinary(60)`?
I'm not familiar with passing a string as the first argument.

Comment (qq1u) [reply models/user.py:44]: String as the first argument, which will be the name of the field in the table.

File: models/user.py [hunk 5]
```diff
@@ -59,10 +58,16 @@ class UserModel(BaseModel):
         collection_class=NobiruList,
     )
 
-    def __init__(self, **kwargs):
-        super(UserModel, self).__init__(**kwargs)
-        self.hash_digest = UserModel._hash_pw(kwargs["password"])
-        self.archived = False
+    @property
+    def password(self):
+        raise AttributeError("password field is not allowed to access")
```
Comment (NickSchimek) [inline models/user.py:63]: I like this better! Good work!

**Review by NickSchimek — APPROVED** [review]
Excellent! Thanks 🙏

**Pull request merged**